crates/ra_ide_api/src/completion/complete_postfix.rs

```diff
@@ -21,28 +22,46 @@ fn postfix_snippet(ctx: &CompletionContext, label: &str, detail: &str, snippet:
         .snippet_edit(edit)
 }
 
+fn is_bool_or_unknown(ctx: &CompletionContext) -> bool {
+    let receiver_ty = ctx.dot_receiver.and_then(|it| ctx.analyzer.type_of(ctx.db, it));
+    if let Some(ty) = receiver_ty {
+        match ty {
+            Ty::Apply(at) => match at.ctor {
+                TypeCtor::Bool => true,
+                _ => false,
```

Comment:
`if (loop {}) {}` is allowed. the `loop {}` has type `!` (`TypeCtor::Never`) here.

Reply:
Oh... you're right, but these are postfix completions... do we really want to suggest `if` for the never type?